core/src/main/java/io/questdb/griffin/engine/groupby/SampleByFillPrevNotKeyedRecordCursor.java

```diff
@@ -106,13 +117,15 @@ public boolean hasNext() {
 
         final int n = groupByFunctions.size();
         for (int i = 0; i < n; i++) {
+            interruptor.checkInterrupted();
             groupByFunctions.getQuick(i).computeFirst(simpleMapValue, baseRecord);
         }
 
         while (base.hasNext()) {
             long timestamp = timestampSampler.round(baseRecord.getTimestamp(timestampIndex));
             if (lastTimestamp == timestamp) {
                 for (int i = 0; i < n; i++) {
+                    interruptor.checkInterrupted();
```

Comment:
this is intense. Function exec should take long time, lets keep this on cursor only